tools/astarte_export/lib/export.ex

```diff
@@ -0,0 +1,283 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2019 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Export do
+  alias Astarte.Export.FetchData
+  require Logger
+  
+  @moduledoc """
+  This  module provide API functions to export realm device 
+  data in a xml format. This data can be used by astarte_import 
+  application utlity  to import into a new realm.
+  """
+  @doc """
+  The export_realm_data/2 function required 2 arguments to export 
+  the realm data into XML format.
+  the arguments are
+   -realm-name -> This is a string format of input
+   - path      -> path where to export the realm file.
+ 
+  @spec export_relam_data(String.t, String.t) :: :ok
+
+  """ 
+ 
+  def export_realm_data(realm, path) do
+    with true <- File.dir?(path) do
+      timestamp = format_time
+      filename  = path <> "/" <> realm <> "_" <> timestamp <>  ".xml"
+      generate_xml(realm, filename)
+    else
+      result -> {:error, :invalid_parameters}
+    end
+  end
+
+  defp generate_xml(realm, file) do
+    {:ok, xmlfile} = File.open(file, [:write])
+    Logger.info("Export started.", realm: realm)
+    xml_data = seralize_xml(realm)
+    Logger.info("XML Seralization completed", realm: realm)
+    :ok = IO.puts( xmlfile, xml_data)
+    Logger.info("Export completed into file: #{file}", realm: realm) 
+    :ok = File.close(xmlfile)
+  end
+
+  defp seralize_xml(realm) do
+    xml_data = 
+      seralize_xml(:astrate, [{:realm, realm}])
```

Comment:
What happens here if I have several GB of data?
Do you keep everything in memory until you write it to disk?

Reply:
Will take of this.. development changes are in progress

Reply:
Working on this.